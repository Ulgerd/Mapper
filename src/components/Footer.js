import React from 'react';
import styled from 'styled-components'

const StyledFooter = styled.div`
  position:absolute;
  left:0;
  bottom:0;
  right:0;
  display: flex;
  flex-direction: column;
  background-color: lightgray;
  padding-bottom:1.5em;
`

export function Footer() {
  return (
    <StyledFooter>
      <span>Код можно посмотреть тут:
        <a href="https://github.com/Ulgerd/Mapper/tree/master/src">
          GitHub Mapper
        </a>
      </span>
      <span>Первая часть квалификационного задания:
        <a href="https://resume.ulgerd.now.sh/FunBox_Level_1.pdf">
          Level_I
        </a>
      </span>
      <span>Резюме:
        <a href="https://resume.ulgerd.now.sh/Pylaev_React_Jun.html">
          Resume
        </a>
      </span>
    </StyledFooter>
  );
}

export default Footer;
