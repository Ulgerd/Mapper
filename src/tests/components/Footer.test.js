import React from 'react';
import { mount } from 'enzyme'
import { Footer } from '../../components/Footer.js';
import renderer from 'react-test-renderer';

describe('Footer component tests',()=>{
    let wrapper

    beforeEach(()=>{
        wrapper = mount(<Footer />)
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('contains div', () => {
        expect(wrapper.find('div')).toHaveLength(1)
    });

    it('contains span', () => {
        expect(wrapper.find('span')).toHaveLength(3)
    });

    it('contains a', () => {
        expect(wrapper.find('a')).toHaveLength(3)
    });

    it('renders correctly', () => {
      const tree = renderer
        .create(<Footer />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});
