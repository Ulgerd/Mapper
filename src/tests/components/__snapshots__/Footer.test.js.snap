// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component tests renders correctly 1`] = `
<div
  className="sc-bdVaJa hgsmGs"
>
  <span>
    Код можно посмотреть тут:
    <a
      href="https://github.com/Ulgerd/Mapper/tree/master/src"
    >
      GitHub Mapper
    </a>
  </span>
  <span>
    Первая часть квалификационного задания:
    <a
      href="https://resume.ulgerd.now.sh/FunBox_Level_1.pdf"
    >
      Level_I
    </a>
  </span>
  <span>
    Резюме:
    <a
      href="https://resume.ulgerd.now.sh/Pylaev_React_Jun.html"
    >
      Resume
    </a>
  </span>
</div>
`;
